contracts/modules/legacy/LegacyPooledStaking.sol

```diff
@@ -1197,208 +1198,196 @@ contract LegacyPooledStaking is IPooledStaking, MasterAwareV2 {
+  ) internal {
+    ( /* uint stakingPoolId */, address stakingPoolAddress) = cover.createStakingPool(
+      migrationData.managerAddress,
+      migrationData.isPrivatePool,
+      migrationData.initialPoolFee,
+      migrationData.maxPoolFee,
+      productInitParams,
+      migrationData.ipfsDescriptionHash
+    );
 
-  struct StakingPoolMigrationData {
-    uint initialPoolFee;
-    uint maxPoolFee;
-    uint[TRANCHE_COUNT] stakerTrancheRatios;
-    uint deposit;
-    address stakerAddress;
-    bool isPrivatePool;
-    string ipfsDescriptionHash;
-    address managerAddress;
+    // TODO - is this ok here or do we want granular approvals in the loop below?
+    token().approve(address(tokenController()), migrationData.deposit);
+
+    uint totalStakeRatio = 0;
+    uint totalStake = 0;
+    uint firstTrancheId = block.timestamp / 91 days + 1;
+
+    for (uint i = 0; i < MAX_ACTIVE_TRANCHES; i++) {
+      uint trancheDeposit = migrationData.deposit * migrationData.trancheStakeRatio[i] / 100;
+      if (trancheDeposit == 0) {
+        continue;
+      }
+
+      totalStakeRatio += migrationData.trancheStakeRatio[i];
+      // Prevent dust due to precision loss and deposit everything remaining in the last tranche
+      if (totalStakeRatio == 100) {
+        trancheDeposit = migrationData.deposit - totalStake;
+      }
+      totalStake += trancheDeposit;
+
+      IStakingPool(stakingPoolAddress).depositTo(
+        trancheDeposit,
+        firstTrancheId + i,
+        type(uint).max,
+        migrationData.managerAddress
+      );
+    }
+
+    // TODO This can be enabled once the foundation deposit migration is done
+    // require(totalStake == migrationData.deposit, "Migrated stake != staker deposit");
   }
 
   function migrateToNewV2Pool(address stakerAddress) external noPendingActions {
-
-    // Addresses marked for implicit migration can be migrated by anyone.
-    // Addresses who are not can only be migrated by calling this function themselves.
+    // Only selected stakers are automatically migrated
     require(
-      stakerAddress == ARMOR || // Armor
-      stakerAddress == HUGH || // Hugh
-      stakerAddress == NEXUS_FOUNDATION, // Foundation
+      stakerAddress == ARMOR_STAKER ||
+      stakerAddress == HUGH ||
+      stakerAddress == NM_FOUNDATION,
 
       "You are not authorized to migrate this staker"
     );
 
+    // Read and set deposit to zero to avoid re-entrancy
+    uint deposit = stakers[stakerAddress].deposit;
+    require(deposit > 0, "Address has no stake to migrate");
+    stakers[stakerAddress].deposit = 0;
 
     INXMToken nxm = token();
     uint nxmBalanceBefore = nxm.balanceOf(address(this));
 
-    // ratios have no decimal points. eg 5 is 5%
-    uint[TRANCHE_COUNT] memory stakerTrancheRatios;
-
-    (ProductInitializationParams[] memory params, uint deposit) = getStakerConfig(stakerAddress);
+    ProductInitializationParams[] memory productInitParams = getProductInitParams(
```

Comment:
I took out the deposit from the output of this function on purpose, because it was/is a getter, but it mutated state.

Reply:
The naming was not reflective of the mutation agreed i think what you did is better.

The usage of it is in only 1 place, it's not public but it does make things easier to reason about to have a pure view function. Thus we can even make it public in case it helps with checks/assertions.

Reply:
made it public